lib/monocle.js

```diff
@@ -20,68 +23,119 @@ Monocle.prototype.getCache = function() {
 
 ['get', 'post', 'put', 'patch', 'delete', 'options'].forEach(function(method) {
     Monocle.prototype[method] = function(path, options) {
-        switch (method) {
-            // Check cache if attempting to get resource
-            case 'get':
-                var cached = this._cache.get(path);
-                if (cached) {
-                    if (!options.props) {
-                        return Promise.resolve(cached);
-                    }
-                    var found = true;
-                    for (var i in options.props) {
-                        if (!cached.hasOwnProperty(options.props[i])) {
-                            this._cache.remove(path);
-                            found = false;
-                            break;
-                        }
-                    }
-                    if (found) return Promise.resolve(cached);
-                }
-                break;
+        return _handle.call(this, method, path, options);
+    };
+});
 
-            // Remove from cache when resource is being updated or removed
-            case 'post':
-            case 'put':
-            case 'delete':
-            case 'patch':
-                this._cache.remove(path);
-                break;
-        }
+var updateBatchTimeout = function() {
+    if (null !== this._batchTimeout) {
+        // Batch timeout already set up, nothing to do.
+        return;
+    }
 
-        var fullPath = buildFullPath(this._base, path);
-        var query = buildQuery(options);
+    this._batchTimeout = setTimeout(function() {
+        processBatch.call(this);
+        this._batchTimeout = null;
+    }.bind(this));
+};
+
+var processBatch = function() {
+    var batched = this._batched;
+    this._batched = [];
+
+    if (1 === batched.length) {
+        var fullPath = buildFullPath(this._base, batched[0].path);
+        var query = buildQuery(batched[0].options);
 
         if (query) fullPath += '?' + query;
 
-        return this._http.request(method.toUpperCase(), fullPath, options)
-        .then(cacheResource.bind(this, method));
-    };
+        return this._http.request(batched[0].method.toUpperCase(), fullPath, batched[0].options)
+        .then(batched[0].resolve)
+        .catch(batched[0].reject);
+    }
+
+    return this._http.request('POST', buildFullPath(this._base, '/_batch'), {
```

Comment:
Other HTTP verbs are supported. The POST here is the wrapper for the batched calls. All batched calls will go over POST, and the internal requests will be wrapped in envelopes that can have their own verbs.